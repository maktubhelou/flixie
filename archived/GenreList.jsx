import React from 'react';
import PropTypes from 'prop-types';
import ListItem from './ListItem';

class GenreList extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            currentGenre: '',
        }
    }
    
    setGenre = (genreName) => {
        this.setState({
            currentGenre: genreName,
        })
    }

    render() {
    const listItems = this.props.genres.map((genres, i) => {
       
        return (
            <ListItem 
            key={i}
            value={genres.name}
            genreName={genres.name}
            setGenre={this.setGenre}
            />
            );
        })

        return listItems;
    }
}

GenreList.propTypes = {
	genres: PropTypes.array,
};

export default GenreList;