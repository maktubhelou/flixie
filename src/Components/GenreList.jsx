import React from 'react'
import PropTypes from 'prop-types'

const styles = {
  listItemStyle: {
    float: 'left',
    maxWidth: '150px',
    border: '1px solid white',
    margin: '2px auto 2px auto',
    padding: '2px 10px',
    color: 'white',
    cursor: 'pointer',
  },
  selectStyle: {
    backgroundColor: 'rgba(255,255,255,0.7)',
    border: '10px solid rgba(255,255,255,0.5)',
  },
}

const GenreList = ({ setGenre, genres = [] }) => (
  <form>
    <select style={styles.selectStyle} onChange={setGenre}>
      {genres.map(genre => {
        const genreName = genre.name
        const genreID = genre.id

        return (
          <option style={styles.listItemStyle} key={genre.id} value={genreID}>
            {genreName}
          </option>
        )
      })}
    </select>
  </form>
)

GenreList.propTypes = {
  setGenre: PropTypes.func.isRequired,
  genres: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
    }),
  ).isRequired,
}

export default GenreList
