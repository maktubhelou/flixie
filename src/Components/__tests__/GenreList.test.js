import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import GenreList from '../GenreList'

describe('<GenreList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<GenreList />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  it('matches snapshot', () => {
    const tree = renderer.create(<GenreList />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
