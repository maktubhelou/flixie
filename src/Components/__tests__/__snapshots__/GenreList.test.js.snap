// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenreList /> matches snapshot 1`] = `
<form>
  <select
    style={
      Object {
        "backgroundColor": "rgba(255,255,255,0.7)",
        "border": "10px solid rgba(255,255,255,0.5)",
      }
    }
  />
</form>
`;
